### PR TITLE
demo(form): improve variant demo

### PR DESCRIPTION
**Pull request by thinkasany**
### 🤔 This is a ...

- [x] 📽️ Demo improvement

### 💡 Background and Solution
增加一个Radio.Group来切换variant 变体，方便预览不同变体的效果
image

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |          demo(form): improve variant demo |
| 🇨🇳 Chinese |     demo(form): 优化 variant demo      |

### CODE REVIEW

File: components/form/demo/variant.tsx [hunk 1]
```diff
@@ -24,78 +26,103 @@ const formItemLayout = {
...
+      {...formItemLayout}
+      onValuesChange={onFormVariantChange}
+      variant={componentVariant}
+      style={{ maxWidth: 600 }}
+      initialValues={{ variant: componentVariant }}
     >
-      <InputNumber style={{ width: '100%' }} />
-    </Form.Item>
+      <Form.Item label="Form variant" name="variant">
+        <Radio.Group>
```
Comment (li-jia-nan) [inline components/form/demo/variant.tsx:44]: 用 Segmented 吧

Comment (thinkasany) [reply components/form/demo/variant.tsx:44]: 我是看到form这边的demo如下面 size/layout/requiredMark  都是用radio.group 想着统一来着，那下面这些要改Segmented 不
image

Comment (li-jia-nan) [reply components/form/demo/variant.tsx:44]: 只改这个就好了

Comment (thinkasany) [reply components/form/demo/variant.tsx]: done 
image

**Review by li-jia-nan — APPROVED** [review]

**Pull request merged**